File: src/constants/schools.js
// Supported campuses at launch.
// emailDomain is used to verify a student's .edu email on sign up
// and to scope listings to only that campus.
export const SCHOOLS = [
  {
    id: 'tufts',
    name: 'Tufts University',
    emailDomain: 'tufts.edu',
  },
  {
    id: 'northeastern',
    name: 'Northeastern University',
    emailDomain: 'northeastern.edu',
  },
];

export function schoolFromEmail(email) {
  const domain = email.split('@')[1]?.toLowerCase();
  return SCHOOLS.find((s) => domain === s.emailDomain) || null;
}

File: backend/app/core/schools.py
SCHOOLS = [
    {"id": "tufts", "name": "Tufts University", "email_domain": "tufts.edu"},
    {"id": "northeastern", "name": "Northeastern University", "email_domain": "northeastern.edu"},
]


def school_from_email(email: str) -> dict | None:
    if "@" not in email:
        return None
    domain = email.rsplit("@", 1)[1].lower()
    return next((school for school in SCHOOLS if school["email_domain"] == domain), None)
